js/tab.js

```diff
@@ -1,13 +1,53 @@
-function openTab(evt, cityName){
-    var i, tabcontent, tablinks;
+//The manageHash function deals with the content changes on hitting a url with a particular tab id
+function manageHash(defaultID){
+    let tabcontent, tablinks;
+    //find contentId(id of the div, needed to be displayed) from the url 
+    let contentId = window.location.hash.substring(1)
+
+    //if no hash(id of div/tab) is given, take first tab as default tab
+    contentId = contentId === ''?defaultID:contentId
+
+    //On click on a tab, usually all the divisions go hidden except the content with targented id name 
+    //converting all the content div display to none
```

Comment:
IOW, close them and just leave the headers? Or make them completely invisible? Not sure what this does.

Reply:
When you set the `display` attribute of an object in the DOM, the browser skips rendering it on the page.  None of the attributes, sizing, etc. of that object affect the layout of the page at all.  The object is still in the DOM, but just completely hidden.  When the `display` attribute of an object is not `none`, but some other value, the browser takes action rendering it as directed.  See https://www.w3schools.com/jsref/prop_style_display.asp.